# Test StackingClassifier:

#### 1. Use the breast-bin.csv dataset

In [1]:
from si.io.csv_file import read_csv

In [2]:
df_path = r"C:\Users\Mariana\si\datasets\breast-bin\breast-bin.csv"
df = read_csv(df_path, sep = ',', features = True, label = True) 

#### 2. Split the data into train and test sets

In [3]:
from si.model_selection.split import train_test_split

In [4]:
train_dataset, test_dataset = train_test_split(df, test_size=0.33, random_state=42)

In [5]:
print(f'(As dimensões dos dados de treino correspondem a:', train_dataset.shape())
print(f'(As dimensões dos dados de teste correspondem a:',test_dataset.shape())

(As dimensões dos dados de treino correspondem a: (468, 9)
(As dimensões dos dados de teste correspondem a: (230, 9)


Os dados de treino apresentam 468 linhas e 9 colunas, enquanto que os dados de teste apresentam 230 linhas e 9 colunas.

#### 3. Create a KNNClassifier model

In [6]:
from si.models.knn_classifier import KNNClassifier
knn_model = KNNClassifier(k=3)

#### 4. Create a LogisticRegression model

In [7]:
from si.models.logistic_regression import LogisticRegression
logistic_regression_model = LogisticRegression(l2_penalty=1, alpha=0.001, max_iter=1000)

#### 5. Create a DecisionTree model

In [8]:
from si.models.decision_tree_classifier import DecisionTreeClassifier
decision_tree_model = DecisionTreeClassifier(max_depth=5)

#### 6. Create a second KNNClassifier model (final model)

In [9]:
knn_model_2 = 

#### 7. Create a Stacking Classifier model using the previous classifiers. The second KNNClassifier model must be used as the final model.

In [ ]:
from si.ensemble.stacking_classifier import StackingClassifier
#stacking_classifier_model = StackingClassifier(train_dataset)

#### 8. Train the StackingClassifier model. What is the score of the model on the test set?

In [17]:
#stacking_classifier_model = StackingClassifier(train_dataset)

In [16]:
score = stacking_classifier_model.score(test_dataset)
print(f"Score: {score}")

TypeError: 'Dataset' object is not iterable